src/rgw/rgw_sal_motr.cc

```diff
@@ -2134,11 +2132,15 @@ int MotrObject::read_mobj(const DoutPrefixProvider* dpp, int64_t off, int64_t en
 
   left = end - off;
   for (; left > 0; off += actual) {
-    if (left < bs)
-      bs = this->get_optimal_bs(left);
     actual = bs;
-    if (left < bs)
+    if (left < bs){
+      mobj->ob_entity.en_flags |= M0_OOF_HOLE;
```

Comment:
move this below if block, outside the if block.

Reply:
Done